src/jit/inlinepolicy.cpp

```diff
@@ -458,30 +464,132 @@ double LegacyPolicy::determineMultiplier()
         JITDUMP("\nInline candidate has const arg that feeds a conditional.  Multiplier increased to %g.", multiplier);
     }
 
-    return multiplier;
-}
+    switch (inlCallsiteFrequency)
+    {
+    case InlineCallsiteFrequency::RARE:
+        // Note this one is not additive, it uses '=' instead of '+='
+        multiplier = 1.3;
+        JITDUMP("\nInline candidate callsite is boring.  Multiplier limited to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::BORING:
+        multiplier += 1.3;
+        JITDUMP("\nInline candidate callsite is boring.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::WARM:
+        multiplier += 2.0;
+        JITDUMP("\nInline candidate callsite is warm.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::LOOP:
+        multiplier += 3.0;
+        JITDUMP("\nInline candidate callsite is in a loop.  Multiplier increased to %g.", multiplier);
+        break;
+    case InlineCallsiteFrequency::HOT:
+        multiplier += 3.0;
+        JITDUMP("\nInline candidate callsite is hot.  Multiplier increased to %g.", multiplier);
+        break;
+    default:
+        assert(!"Unexpected callsite frequency");
+        break;
+    }
 
-//------------------------------------------------------------------------
-// hasNativeCodeSizeEstimate: did this policy estimate native code size
-//
-// Notes: 
-//    Temporary scaffolding for refactoring work.
+#ifdef DEBUG
 
-bool LegacyPolicy::hasNativeSizeEstimate()
-{
-    return (inlStateMachine != nullptr);
+    int additionalMultiplier = JitConfig.JitInlineAdditionalMultiplier();
+
+    if (additionalMultiplier != 0)
+    {
+        multiplier += additionalMultiplier;
+        JITDUMP("\nmultiplier increased via JitInlineAdditonalMultiplier=%d to %g.", additionalMultiplier, multiplier);
+    }
+
+    if (inlCompiler->compInlineStress())
```

Comment:
I'm seeing another instance that cab replaced by this.
https://github.com/dotnet/coreclr/blob/master/src/jit/importer.cpp#L76